### PR TITLE
Fix flatten with pointers containing escaped chars

### PR DESCRIPTION
**Pull request by fredbi**
Fix flatten with pointers containing escaped chars (e.g. '{}')

Fixed undue duplicate name detection when resolving remote refs

Improves behavior in edge cases where schemas points to responses
or parameters.

This makes the flatten behavior consistent with swagger validate.

* now unmarshalls the struct as a schema: if this works, flattening
moves on, with a warning to the user (as this is valid but most
likely unintended)
* improves go-swagger/go-swagger#1602 and go-swagger/go-swagger#1614
* contributes go-swagger/go-swagger#1621
* Expand schemas when $ref'ed to (shared) responses or parameters

### CODE REVIEW

**Comment by gregmarr** [comment]
Are you saying that we broke compatibility with 1.7 or with pre-1.7.  This says that we support the last two versions, but that our CI runs 1.7.

https://goswagger.io/faq/faq_setup.html#what-is-the-minimal-go-version-required

So we only officially support 1.9 and later.

**Comment by casualjim** [comment]
yes only last 2 versions

**Comment by fredbi** [comment]
OK I remove this CI test.
I am just adding the comment that this package remains compatible with pre go1.8.
This is just the dependency which breaks this. So this package may be used with pre1.8 
provided one fetches github.com/asaskevich/govalidator up to v9

File: internal/pre_go18.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// +build !go1.8
```
Comment (gregmarr) [inline internal/pre_go18.go]: We don't need to support pre-1.8.

**Pull request merged**